contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -576,15 +593,19 @@ int process_add_smartcard_key(struct sshbuf* request, struct sshbuf* response, s
 		is_reg_sub_key_exists(user_root, SSH_PKCS11_PROVIDERS_ROOT, canonical_provider))
 		goto done;
 
-	pkcs11_init(0);
-
 	count = pkcs11_add_provider(canonical_provider, pin, &keys, NULL);
 	if (count <= 0) {
 		debug("failed to add key to store");
 		goto done;
 	}
 	for (i = 0; i < count; i++) {
 		key = keys[i];
+		if (sa.lpSecurityDescriptor)
+			LocalFree(sa.lpSecurityDescriptor);
+		if (reg)
+			RegCloseKey(reg);
+		if (sub)
+			RegCloseKey(sub);
 		memset(&sa, 0, sizeof(SECURITY_ATTRIBUTES));
```

Comment:
Should we zero `reg` and `sub` to avoid a potential second `RegCloseKey()` in the `done` path?

Reply:
Absolutely! Thanks.